test/irmin-pack/test_inode.ml

```diff
@@ -147,14 +223,88 @@ let test_remove_inodes () =
       ]
   in
   let v5 = Inode.Val.remove v4 "a" in
-  let h1 = Private.hash v1 in
-  let h5 = Private.hash v5 in
-  check_hash
-    "v1 and v5 have different internal representations but the same hash" h1 h5;
+  check_values "node x+y+z obtained two ways" v1 v5;
   Alcotest.(check bool) "v1 stable" (Private.stable v1) true;
   Alcotest.(check bool) "v5 stable" (Private.stable v5) true;
   Context.close t
 
+(** [steps] is a list of length 8 containing the necessary steps to fill a tree
+    of depth=3.
+
+    [trees] is a list of length 2^8 containing the [Node.Val.t] formed from the
+    powerset of [steps] and constructed using [Node.Val.v]. [trees] contains the
+    empty tree and the full 8-leaves tree.
+
+    From all the 2^8 trees, assert that independently, the 8 possible
+    [add]/[remove] operations yield a tree with its representation included in
+    [trees]. *)
+let test_representation_uniqueness_maxdepth_3 () =
+  let steps = gen_steps 3 in
+  let content_per_step =
+    List.map (fun s -> (s, H_contents.hash s |> normal)) steps
+    |> StepMap.of_list
+  in
+  let steps_per_tree : StepSet.t list =
+    powerset steps |> List.map StepSet.of_list
+  in
+  Alcotest.(check int) "Size of the powerset" (List.length steps_per_tree) 256;
+  let trees : Inode.value list =
+    List.map
+      (fun steps ->
+        let steps = StepSet.elements steps in
+        let contents =
+          List.map (fun s -> StepMap.find s content_per_step) steps
```

Comment:
it's maybe a bit clearer if you generate the content hash here, instead of looking for it in `content_per_step`?

Reply:
It is the `content` and not the `content hash` that I need to get here.

I can't generate the `content` directly here because I need to be able to fetch it from a step when testing the `add` function below.